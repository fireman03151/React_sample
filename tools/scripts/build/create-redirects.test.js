/* global describe expect it */

const { createRedirects } = require('./create-redirects');

const testLocations = {
  api: 'https://api.example.com',
  newsProxy: 'https://news.example.com',
  forumProxy: 'https://forum.example.com'
};

describe('createRedirects', () => {
  it('is a function', () => {
    expect(typeof createRedirects).toEqual('function');
  });

  it('returns a string', () => {
    expect(typeof createRedirects(testLocations)).toEqual('string');
  });

  it('replaces instances of `#{{...}}` with the locations provided', () => {
    expect.assertions(4);

    const apiPlaceholderRE = /#\{\{API\}\}/;
    const newsPlaceholderRE = /#\{\{NEWS\}\}/;
    const forumPlaceholderRE = /#\{\{FORUM\}\}/;
    const redirects = createRedirects(testLocations);

    const hasApiPlaceholder = apiPlaceholderRE.test(redirects);
    const hasNewsPlaceholder = newsPlaceholderRE.test(redirects);
    const hasForumPlaceholder = forumPlaceholderRE.test(redirects);

    expect(hasApiPlaceholder).toBe(false);
    expect(hasNewsPlaceholder).toBe(false);
    expect(hasForumPlaceholder).toBe(false);

    const { forumProxy } = testLocations;
    expect(redirects.includes(`${forumProxy}`)).toBe(true);
  });

  it('throws when any location is missing', () => {
    expect.assertions(3);

    const api = 'api';
    const newsProxy = 'newsProxy';
    const forumProxy = 'forumProxy';

    const noApi = { forumProxy, newsProxy };
    const noNews = { api, forumProxy };
    const noForum = { api, newsProxy };

    expect(() => createRedirects(noApi)).toThrow();
    expect(() => createRedirects(noNews)).toThrow();
    expect(() => createRedirects(noForum)).toThrow();
  });

  it('matches the snapshot', () =>
    expect(createRedirects(testLocations)).toMatchSnapshot());
});
