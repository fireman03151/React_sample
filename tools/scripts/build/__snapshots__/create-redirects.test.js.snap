// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createRedirects matches the snapshot 1`] = `
"
#
# WARNING: The order of this list is very important.
#

# domain mappings
https://freecodecamp-dev.netlify.com/*        https://www.freecodecamp.dev/:splat 301!
https://freecodecamp-org.netlify.com/*        https://www.freecodecamp.org/:splat 301!

# pages
/about                                        https://news.example.com/about 200
/academic-honesty                             https://news.example.com/academic-honesty 200
/code-of-conduct                              https://news.example.com/code-of-conduct 200
/copyright                                    https://news.example.com/copyright-policy 200
/copyright-policy                             https://news.example.com/copyright-policy 200
/privacy                                      https://news.example.com/privacy-policy 200
/privacy-policy                               https://news.example.com/privacy-policy 200
/shop                                         https://news.example.com/shop 200
/software-resources-for-nonprofits            https://news.example.com/software-resources-for-nonprofits 200
/sponsors                                     https://news.example.com/sponsors 200
/support                                      https://news.example.com/support 200
/terms                                        https://news.example.com/terms-of-service 200
/terms-of-service                             https://news.example.com/terms-of-service 200

# applications
/forum/*                                      https://forum.example.com/:splat 200
/news                                         https://news.example.com/ 200
/news/*                                       https://news.example.com/:splat 200

# auth redirects
/deprecated-signin                            https://api.example.com/signin 200!
/email-signin                                 https://api.example.com/signin 200!
/login                                        https://api.example.com/signin 200!
/logout                                       https://api.example.com/signout 200!
/passwordless-change                          https://api.example.com/confirm-email 200!
/signin                                       https://api.example.com/signin 200!
/signup                                       https://api.example.com/signin 200!

# certification redirects
/:username/front-end-certification                     /certification/:username/legacy-front-end 301
/:username/data-visualization-certification            /certification/:username/legacy-data-visualization 301
/:username/back-end-certification                      /certification/:username/legacy-back-end 301
/:username/information-security-and-quality-assurance  /certification/:username/legacy-information-security-and-quality-assurance 301
/:username/full-stack-certification                    /certification/:username/full-stack 301

# unsubscribe redirects
/u/*                                          https://api.example.com/u/:splat 200!
/ue/*                                         https://api.example.com/ue/:splat 200!
/unsubscribe/*                                https://api.example.com/unsubscribe/:splat 200!

# misc redirects
/agile                                        / 301
/all-stories                                  / 301
/chat                                         https://gitter.im/FreeCodeCamp/FreeCodeCamp 301
/donation/settings                            /donate 302
/field-guide/*                                / 301
/how-nonprofit-projects-work                  / 301
/learn-to-code                                /learn 301
/map                                          /learn 200
/nonprofit-project-instructions               / 301
/Nonprofits                                   / 301
/nonprofits                                   / 301
/nonprofits-form                              / 301
/pmi-acp-agile-project-managers               / 301
/pmi-acp-agile-project-managers-form          / 301
/stories                                      / 301
/twitch                                       https://twitch.tv/freecodecamp 301
/welcome                                      / 200

"
`;
